src/cli/cli.cpp

```diff
@@ -602,6 +612,259 @@ void Interpreter::ProcessChildTimeout(int argc, char *argv[])
     AppendResult(error);
 }
 
+#if OPENTHREAD_ENABLE_APPLICATION_COAP
+
+void Interpreter::ProcessCoapServer(int argc, char *argv[])
+{
+    ThreadError error = kThreadError_None;
+
+    VerifyOrExit(argc > 0, error = kThreadError_InvalidArgs);
+
+    if (strcmp(argv[0], "start") == 0)
+    {
+        SuccessOrExit(error = otCoapServerStart(mInstance));
+        SuccessOrExit(error = otCoapServerAddResource(mInstance, &mResource));
+        sServer->OutputFormat("Starting server and adding resource '%s' ... ", mResource.mUriPath);
+    }
+    else if (strcmp(argv[0], "stop") == 0)
+    {
+        otCoapServerRemoveResource(mInstance, &mResource);
+        SuccessOrExit(error = otCoapServerStop(mInstance));
+        sServer->OutputFormat("Stopping server ... ");
+    }
+    else if (strcmp(argv[0], "name") == 0)
+    {
+        if (argv[1]){
+            mResource.mUriPath = argv[1];
```

Comment:
Due to the definition mResource is persistent and therefore mUriPath as well. At least it works as supposed?